proto/speechly/slu/v2beta1/batch.proto

```diff
@@ -0,0 +1,200 @@
+// all languages or on all Payment Plans.
+message ProcessingConfiguration {
+  // The processing should include the token level transcription and determination of time stamps for tokens.
+  bool tokenize = 1;
+  // The processing should include translating the audio to English.
+  bool translate = 2;
+  // The processing should not include transcribing the audio to the source language. This option should be used
+  // with translate (or other similar option) to suppress the normal Speech Recognition processing.
+  bool skip_transcribe = 3;
+}
+
+// Describes a single batch operation.
+message Operation {
+  // The status of the operation.
+  enum Status {
+    // Default status is empty.
+    STATUS_INVALID = 0;
+    // The operation is queued for processing.
+    STATUS_PENDING = 1;
+    // The operation is complete and transcript is available.
+    STATUS_DONE = 2;
+    // The processing failed. Error reason is available.
+    STATUS_ERROR = 3;
+  }
+  // The id of the operation.
+  string id = 1;
+  // The reference id of the operation, if given.
+  string reference = 2;
+  // The id of the batch the operation belongs to.
+  string batch_id = 11;
+  // The reference id of the operation, if given.
+  string batch_reference = 12;
+  // The current status of the operation.
+  Status status = 3;
+  // The language code of the detected language.
+  string language_code = 4;
+  // The application context for the operation.
+  string app_id = 5;
+  // If the operation status is STATUS_DONE and the destination is not set,
+  // the results of the processing.
+  OperationResult result = 6;
+  // The duration of the audio.
+  google.protobuf.Duration duration = 7;
+  // Contains a machine readable error type if the operation status is
+  // STATUS_ERROR.
+  enum OperationError {
+    OPERATION_ERROR_INVALID = 0;
+    // The input was in a language that was not supported in this context.
+    OPERATION_ERROR_UNSUPPORTED_LANGUAGE = 1;
+    // An internal error occurred.
+    OPERATION_ERROR_INTERNAL = 2;
+    // The source could not be read or understood.
+    OPERATION_ERROR_INVALID_SOURCE = 4;
+    // The results could not be written to the destination.
+    OPERATION_ERROR_INVALID_DESTINATION = 5;
+    // The provided audio was not in a supported format.
+    OPERATION_ERROR_INVALID_AUDIO = 6;
+  }
+  OperationError error_code = 13;
+  // Contains a human readable description of the error if the operation status is
+  // STATUS_ERROR.
+  string error_description = 14;
+  // The locator to the source audio.
+  string source_url = 15;
+  // The locator to the result target.
+  string destination_url = 16;
+}
+
+// Describes the results of the processing that took place.
+message OperationResult {
+  // The various transcripts describing the content of the processed audio.
+  repeated Transcript transcripts = 1;
+}
+
+// Describes the content of an audio as text.
+message Transcript {
```

Comment:
Maybe `Transcript` is unnecessary, as it only wrapped in `OperationResult`? Instead the `Operation` could include `repeated OperationResult` which could have `ResultType` instead of `TranscriptType`. This would make the result tree shallower and thus easier to understand.

Reply:
I added the `OperationResult` as it is expected that there will be result types that are sufficiently different from `Transcript` in the near future, so they can be added with only additive changes.

Reply:
that would be possible also by extending the `OperationResult` itself, it could be just
```
message OperationResult {
  enum ResultType {
    ...
    RESULT_TYPE_TRANSCRIPT_LEXICAL = 1;
    ..
  }
  ResultType type = 1;
  string text = 2;
  repeated Token tokens = 3;
  // future additions here
  // repeated Event events = 4;
}
```
This would collect the result alternatives in the same type, instead of an additional intermediate container type.